stl/inc/algorithm

```diff
@@ -1263,6 +1263,43 @@ namespace ranges {
 } // namespace ranges
 #endif // __cpp_lib_concepts
 
+#ifdef __cpp_lib_concepts
+
+namespace ranges {
+    // ALIAS TEMPLATE copy_result
+    template <class _In, class _Out>
+    using move_result = in_out_result<_In, _Out>;
+
+    class _Move_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        template <input_iterator _It, sentinel_for<_It> _Se, weakly_incrementable _Out>
+        requires indirectly_movable<_It, _Out> constexpr move_result<_It, _Out> operator()(
+            _It _First, _Se _Last, _Out _Result) const {
+            _Adl_verify_range(_First, _Last);
+            auto _UFirst      = _Get_unwrapped(_STD move(_First));
+            const auto _ULast = _Get_unwrapped(_STD move(_Last));
+            for (; _UFirst != _ULast; ++_UFirst, (void) ++_Result) {

```

Comment:
I believe we would want to have a lot of specializations for the different kinds of types / iterators.

You could have a look at std::copy